### PR TITLE
Guard against memory corruption in Space::processResets()

### PR DESCRIPTION
**Pull request by hyperlogic**
Check the proxyID before reading from the _proxies vector and writing into the _owners vector.

### CODE REVIEW

**Comment by hyperlogic** [comment]
QA test:
  * switch between domains frequently.

Expected result:
  You should not crash.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10983/HighFidelity-Beta-PR13499-9772fc4.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10983/HighFidelity-Beta-PR13499-9772fc4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10983/HighFidelity-Beta-Interface-PR13499-9772fc4.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10988/HighFidelity-Beta-PR13499-17f9a01.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10988/HighFidelity-Beta-PR13499-17f9a01.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10988/HighFidelity-Beta-Interface-PR13499-17f9a01.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10988/HighFidelity-Beta-PR13499-17f9a01.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10988/HighFidelity-Beta-Interface-PR13499-17f9a01.exe

File: libraries/workload/src/workload/Space.cpp [hunk 1]
```diff
@@ -44,6 +44,11 @@ void Space::processResets(const Transaction::Resets& transactions) {
     for (auto& reset : transactions) {
         // Access the true item
         auto proxyID = std::get<0>(reset);
+
+        // Guard against proxyID being past the end of the list.
+        if (proxyID < 0 || proxyID >= (int32_t)_proxies.size() || proxyID >= (int32_t)_owners.size()) {
```
Comment (samcake) [inline libraries/workload/src/workload/Space.cpp]: Actually,  the check on Owners doesn;t need to be done

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Did not crash

**Pull request merged**